Tests/Unit/Admin/Extension/PublishTimePeriodExtensionTest.php

```diff
@@ -34,6 +36,26 @@ public function testFormMapper()
             ->method('add')
             ->will($this->returnSelf());
 
+        $builder = $this->getMockBuilder(FormBuilderInterface::class)->getMock();
+        $builder->expects($this->exactly(2))
+            ->method('get')
+            ->withConsecutive(['publish_start_date'], ['publish_end_date'])
+        ;
+        $formFactory = $this->getMockBuilder(FormFactoryInterface::class)->getMock();
+        $formFactory->expects($this->once())
+            ->method('createBuilder')
+            ->will($this->returnValue($builder))
+        ;
```

Comment:
You should avoid making objects both a *mock* and a *stub* in automated tests.

 * **Mocks** define exceptations for command methods that should be matched in order for the test to pass (`->expects($this->once())`)
 * **Stubs** are query methods that just return values required for running the test (`->will($this->returnValue($builder))`)

In short, the method defined on line 40 is a mock: `->get()` has to be called 2 times with both field names. Yet, the method defined here is a stub: Calling `createBuilder()` always returns the builder.

So please change `$this->once()` to `$this->any()` on line 45 (same applies to methods below)